frontends/main/src/app/certificate/[certificateType]/[uuid]/page.tsx

```diff
@@ -1,16 +1,69 @@
 import React from "react"
+import { Metadata } from "next"
 import CertificatePage from "@/app-pages/CertificatePage/CertificatePage"
 import { prefetch } from "api/ssr/prefetch"
 import { certificateQueries } from "api/mitxonline-hooks/certificates"
 import { HydrationBoundary } from "@tanstack/react-query"
 import { isInEnum } from "@/common/utils"
 import { notFound } from "next/navigation"
+import { standardizeMetadata } from "@/common/metadata"
+import { courseCertificatesApi, programCertificatesApi } from "api/mitxonline"
+import * as Sentry from "@sentry/nextjs"
+
+const { NEXT_PUBLIC_ORIGIN } = process.env
 
 enum CertificateType {
   Course = "course",
   Program = "program",
 }
 
+export async function generateMetadata({
+  params,
+}: PageProps<"/certificate/[certificateType]/[uuid]">): Promise<Metadata> {
+  const { certificateType, uuid } = await params
+
+  let title, displayType, userName
+
+  try {
+    if (certificateType === CertificateType.Course) {
+      const { data } = await courseCertificatesApi.courseCertificatesRetrieve({
+        cert_uuid: uuid,
+      })
+
+      title = data.course_run.course.title
+
+      displayType = "Module Certificate"
+
+      userName = data?.user?.name
+    } else {
+      const { data } = await programCertificatesApi.programCertificatesRetrieve(
+        {
+          cert_uuid: uuid,
+        },
+      )
+
+      title = data.program.title
+
+      displayType = `${data.program.program_type} Certificate`
+
+      userName = data.user.name
+    }
+  } catch (error) {
+    Sentry.captureException(error)
+    console.error("Error fetching certificate for metadata", {
+      certificateType,
+      uuid,
+      error,
+    })
+    return standardizeMetadata({})
+  }
+
+  return standardizeMetadata({
+    title: `${userName}'s ${displayType}`,
+    description: `${userName} has successfully completed the Universal Artificial Intelligence ${displayType}: ${title}`,
```

Comment:
Can you give me an example of how this renders? 

We have to be careful about hard-coding "Universal Artificial Intelligence" because this won't stay forever. In a case like this, which isn't very visible, I would just drop "Universal Artificial Intelligence" altogether for better future-compatibility. 

```suggestion
    description: `${userName} has successfully completed the  ${displayType}: ${title}`,
```

Reply:
That currently looks like: "Joe Learner has successfully completed the Universal Artificial Intelligence Module Certificate: The Course Name".

With your suggested change: "Joe Learner has successfully completed the Module Certificate: The Course Name".

`displayType` is either "Module Certificate", "Series Certificate" or "MicroMasters® Certificate".

We do have "Universal Artificial Intelligence" hardcoded on the certificate text itself ("has successfully completed all requirements of the Universal Artificial Intelligence module").